tts/aws/src/lib.rs

```diff
@@ -0,0 +1,1153 @@
+    }
+}
+
+struct AwsPollyComponent;
+
+impl AwsPollyComponent {
+    const ACCESS_KEY_ENV_VAR: &'static str = "AWS_ACCESS_KEY_ID";
+    const SECRET_KEY_ENV_VAR: &'static str = "AWS_SECRET_ACCESS_KEY";
+    const REGION_ENV_VAR: &'static str = "AWS_REGION";
+    const SESSION_TOKEN_ENV_VAR: &'static str = "AWS_SESSION_TOKEN";
+
+    fn create_client() -> Result<AwsPollyTtsApi, TtsError> {
+        with_config_key(Self::ACCESS_KEY_ENV_VAR, Err, |access_key_id| {
+            with_config_key(Self::SECRET_KEY_ENV_VAR, Err, |secret_access_key| {
+                let region =
+                    std::env::var(Self::REGION_ENV_VAR).unwrap_or_else(|_| "us-east-1".to_string());
+                let session_token = std::env::var(Self::SESSION_TOKEN_ENV_VAR).ok();
+
+                AwsPollyTtsApi::new(
+                    access_key_id.to_string(),
+                    secret_access_key.to_string(),
+                    region,
+                    session_token,
+                )
+            })
+        })
+    }
+}
+
+impl VoicesGuest for AwsPollyComponent {
+    type Voice = PollyVoiceImpl;
+    type VoiceResults = PollyVoiceResults;
+
+    fn list_voices(filter: Option<VoiceFilter>) -> Result<VoiceResults, TtsError> {
+        let client = Self::create_client()?;
+        let params = voice_filter_to_describe_params(filter.clone());
+
+        let response = client.describe_voices(params)?;
+        trace!(
+            "AWS describe_voices returned {} voices",
+            response.voices.len()
+        );
+
+        let voice_infos: Vec<VoiceInfo> = response
+            .voices
+            .into_iter()
+            .enumerate()
+            .map(|(i, voice)| {
+                trace!("Converting voice {}: {} ({})", i, voice.name, voice.id);
+                aws_voice_to_voice_info(voice)
+            })
+            .collect();
+
+        trace!("Converted to {} VoiceInfo objects", voice_infos.len());
+        let total_count = Some(voice_infos.len() as u32);
+        Ok(VoiceResults::new(PollyVoiceResults::new(
+            voice_infos,
+            total_count,
+            response.next_token,
+            client,
+            filter,
+        )))
+    }
+
+    fn get_voice(voice_id: String) -> Result<Voice, TtsError> {
+        let client = Self::create_client()?;
+        let response = client.describe_voices(None)?;
+
+        let voice_data = response
+            .voices
+            .into_iter()
+            .find(|v| v.id == voice_id)
+            .ok_or_else(|| TtsError::VoiceNotFound(voice_id.clone()))?;
+
+        Ok(Voice::new(PollyVoiceImpl::new(voice_data, client)))
+    }
+
+    fn search_voices(
+        query: String,
+        filter: Option<VoiceFilter>,
+    ) -> Result<Vec<VoiceInfo>, TtsError> {
+        let client = Self::create_client()?;
+        let params = voice_filter_to_describe_params(filter);
+
+        let response = client.describe_voices(params)?;
+        let voice_infos: Vec<VoiceInfo> = response
```

Comment:
You are doing this post-filtering based on query here but not in the other providers. So mentioning it here, but it's in general about all `search_voices` implementations:

Is there any difference between `query` and the `search-query` in `VoiceFilter`? 

If not, then let's just remove the whole `query` parameter and only have the filter. If there is a difference then make sure all providers are handling it the same, and explain what it means exactly in a doc comment.